src/database/DataTypes/TransactionBatch.js

```diff
@@ -55,7 +72,13 @@ export class TransactionBatch extends Realm.Object {
     if (!this.costPrice) return 0;
     return this.costPrice * this.numberOfPacks;
   }
-
+  /**
+   * removes associated item batch
+   * @param  {Realm} database   App wide database
+   */
+  removeItemBatch(database) {
+    database.delete('ItemBatch', this.itemBatch);
```

Comment:
Agree, although it's only used in ExternalSupplierInvoices (for the time being) it may cause problems if someone careless uses this function in other transactions:
Solitions:
* rename to associate with external supplier invoices item batch and check that parent transaction isExternalSI (since itemBatch is created by externalSI, it's very safe)
* check of totalQuantity 0 ? is it ok to remove empty batches ?
* (best solution) when creating new line, do not create itemBatch, and in transaction.confirm(), check if itemBatch exists, if not create. (this will require, item code to be in transactionBatch, thus migration and big rethink of createRecord because of back and forward relationships between itemBatch, transactionItem, item and transactionBatch.. by the way why so many parent <-> child relationships in mobile ? *note 1.. and also transaction.confirm will start to become a bit of a monster, especially with your comments re pack size 1 translation being in transaction.confirm)

*note 1- instead of child parent back and forth, can we somehow store or access database object within Realm.Object ?, then we can eventually move away from 'circular' relationships, store reference by id and still be able to use getters of parents|children via database.objects(type).filtered(by-id)